### ISSUE TITLE
NuGet Packager not picking up environment variable for -version in VSTS task

### ISSUE THREAD

**Issue by jdshkolnik** [Microsoft/vsts-tasks#2354]
Hi,

If I put in GitVersion.NuGetVerisonV2 it will come up blank. If I put in GitVersion_NuGetVersionV2 it will work as expected.

**Comment by brunobertechini** [comment]
+1

**Comment by brunobertechini** [comment]
I tried:

GitVersion.NuGetVerisonV2
$(GitVersion.NuGetVerisonV2)
"GitVersion.NuGetVerisonV2"

None of them works

But, if I set this feature to OFF (Automatic Versioning)

and use this nuget arguments (in advanced group)

-Version $(GitVersion.NuGetVerisonV2)

it works fine

Bruno

**Comment by zjrunner** [comment]
@brunobertechini - Sorry it took me so long to get to this.  In general if you define a variable with a period in it then you are going to use it in-task with an underscore instead:
https://www.visualstudio.com/en-us/docs/build/define/variables#
So you would use GitVersion_NuGetVersionV2 if it was a variable you defined that gets expanded at task time.  When you moved to -Version $(xxx) then that was expanded by the agent itself, not the build task, so periods are maintained in that case.  

So in short, if you are filling in the EnvironmentVariable field, it is the name of the environment variable and we discover it when running the task.  If you use $(xxx) then by the time our task sees it, it is already a resolved number and not a variable at all.  We take a name explicitly so you have the option of setting and exporting variables from previous tasks.
